File: 2016/FaseLocalUCM/Temperaturas extremas/Temperaturas.c
#include <stdio.h>

int main()
{
  int nInput;
  scanf("%i", &nInput);
  int i;
  for (i = 0; i < nInput; i++)
    {
      int nTemperaturas;
      scanf("%i", &nTemperaturas);
      int temperaturas[nTemperaturas];
      int j;
      for (j = 0; j < nTemperaturas; j++)
        {
          scanf("%i", &temperaturas[j]);
        }
      int nValles = 0;
      int nPicos = 0;
      for (j = 1; j < nTemperaturas - 1; j++)
        {
          if (temperaturas[j - 1] < temperaturas[j] && temperaturas[j] > temperaturas[j + 1]) {
            nPicos++;
          }
          if (temperaturas[j - 1] > temperaturas[j] && temperaturas[j] < temperaturas[j + 1]) {
            nValles++;
          }
        }
      printf("%i %i\n", nPicos, nValles);
    }
  return 0;
}

File: 2016/FaseLocalUCM/Temperaturas extremas/Temperaturas.java
import java.util.Scanner;
public class Temperaturas {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        for (int i = sc.nextInt(); i > 0; i--) {
            int nValles = 0;
            int nPicos = 0;
            int[] temperaturas = new int[sc.nextInt()];
            for (int j = 0 ; j < temperaturas.length; j++) {
                temperaturas[j] = sc.nextInt();
            }
            for (int j = 1; j < temperaturas.length - 1; j++) {
                if (temperaturas[j - 1] < temperaturas[j] && temperaturas[j] > temperaturas[j + 1])
                    nPicos++;
                if (temperaturas[j - 1] > temperaturas[j] && temperaturas[j] < temperaturas[j + 1])
                    nValles++;
            }
            System.out.println(nPicos + " " + nValles);
        }
    }
}
